### PR TITLE
Add postgreSQL Flexible Server Monitoring Dashboard

### PR DESCRIPTION
**Pull request by JohnJMartins**
**What is this feature?**

This PR adds a dedicated and pre-made dashboard to be embedded in the Azure Monitor data source that is dedicated to monitoring metrics of a PostgreSQL Flexible Server. Along with some basic and varied metric panels at the top of the dashboard that provide insight into total storage, usage, read and write metrics, etc, the dashboard also provides dedicated sections with more in-depth metrics; these include Resource Consumption, Autovacuum metrics, PgBouncer metrics, Enhanced metrics, and Hisotrical metrics. Allmetrics are emitted once every minute and stored for up to 93 days.

**Why do we need this feature?**

This makes it possible and easier for users to keep tabs on metrics for their PostgreSQL Flexible Server usage.

Fixes #

**Special notes for your reviewer:**

### CODE REVIEW

File: public/app/plugins/datasource/azuremonitor/dashboards/postgresFlexibleServer.json [hunk 1]
```diff
@@ -0,0 +1,8487 @@
...
+          "y": 0
+        },
+        "id": 74,
+        "options": {
+          "code": {
+            "language": "plaintext",
+            "showLineNumbers": false,
+            "showMiniMap": false
+          },
+          "content": "# Azure Database for PostgreSQL Flexible Server - Monitoring Dashboard\r\n#### - Each metric is emitted at a 1-minute interval and has up to 93 days of history.\r\n#### - For complete list of metrics, refer [Azure PostgreSQL Monitoring Metrics](https://learn.microsoft.com/en-us/azure/postgresql/flexible-server/concepts-monitoring)",
```
Comment (aangelisc) [inline public/app/plugins/datasource/azuremonitor/dashboards/postgresFlexibleServer.json:117]:
```suggestion
          "content": "# Azure Database for PostgreSQL Flexible Server - Monitoring Dashboard\r\n#### - Each metric is emitted at a 1-minute interval and has up to 93 days of history.\r\n#### - For a complete list of metrics, refer to [Azure PostgreSQL Monitoring Metrics](https://learn.microsoft.com/en-us/azure/postgresql/flexible-server/concepts-monitoring)",
```

File: public/app/plugins/datasource/azuremonitor/dashboards/postgresFlexibleServer.json [hunk 2]
```diff
@@ -0,0 +1,8487 @@
...
+        "timeShift": "1m",
+        "title": "Succeeded Connections",
+        "type": "stat"
+      },
+      {
+        "datasource": {
+          "type": "datasource",
+          "uid": "-- Mixed --"
+        },
+        "description": "Number of total transactions that executed in this database.",
```
Comment (aangelisc) [inline public/app/plugins/datasource/azuremonitor/dashboards/postgresFlexibleServer.json:1700]:
```suggestion
        "description": "Number of total transactions executed in this database.",
```

File: public/app/plugins/datasource/azuremonitor/dashboards/postgresFlexibleServer.json [hunk 3]
```diff
@@ -0,0 +1,8487 @@
...
+          "y": 17
+        },
+        "id": 113,
+        "options": {
+          "code": {
+            "language": "plaintext",
+            "showLineNumbers": false,
+            "showMiniMap": false
+          },
+          "content": "#### **NOTE**: While metrics are stored for 93 days, you can only query (in the Metrics tile) for a maximum of 30 days' worth of data on any single chart. If you see a blank chart or your chart displays only part of metric data, verify that the difference between start and end dates in the time picker doesn't exceed the 30-day interval. After you've selected a 30-day interval, you can pan the chart to view the full retention window.",
```
Comment (aangelisc) [inline public/app/plugins/datasource/azuremonitor/dashboards/postgresFlexibleServer.json:2859]:
```suggestion
          "content": "#### **NOTE**: While metrics are stored for 93 days, you can only query (in the Metrics tile) for a maximum of 30 days worth of data on any single chart. If you see a blank chart or your chart displays only part of metric data, verify that the difference between start and end dates in the time picker doesn't exceed the 30-day interval. After you've selected a 30-day interval, you can pan the chart to view the full retention window.",
```

File: public/app/plugins/datasource/azuremonitor/dashboards/postgresFlexibleServer.json [hunk 4]
```diff
@@ -0,0 +1,8487 @@
...
+            },
+            "transparent": true,
+            "type": "text"
+          },
+          {
+            "datasource": {
+              "type": "grafana-azure-monitor-datasource",
+              "uid": "${ds}"
+            },
+            "description": "Overall state of the back ends",
```
Comment (aangelisc) [inline public/app/plugins/datasource/azuremonitor/dashboards/postgresFlexibleServer.json:6491]:
```suggestion
            "description": "Overall state of the backends",
```

File: public/app/plugins/datasource/azuremonitor/dashboards/postgresFlexibleServer.json [hunk 5]
```diff
@@ -0,0 +1,8487 @@
...
+            ],
+            "title": "Sessions by State",
+            "type": "timeseries"
+          },
+          {
+            "datasource": {
+              "type": "grafana-azure-monitor-datasource",
+              "uid": "${ds}"
+            },
+            "description": "Sessions by the type of event for which the back end is waiting",
```
Comment (aangelisc) [inline public/app/plugins/datasource/azuremonitor/dashboards/postgresFlexibleServer.json:6675]:
```suggestion
            "description": "Sessions by the type of event for which the backend is waiting",
```

File: public/app/plugins/datasource/azuremonitor/dashboards/postgresFlexibleServer.json [hunk 6]
```diff
@@ -0,0 +1,8487 @@
...
+          "y": 74
+        },
+        "id": 55,
+        "panels": [
+          {
+            "datasource": {
+              "type": "grafana-azure-monitor-datasource",
+              "uid": "${ds}"
+            },
+            "description": "Indicates if the database is up or not in last 30 days",
```
Comment (aangelisc) [inline public/app/plugins/datasource/azuremonitor/dashboards/postgresFlexibleServer.json:7973]:
```suggestion
            "description": "Indicates if the database is up or not in the last 30 days",
```

**Review by aangelisc — CHANGES_REQUESTED** [review]
Thank you for the contribution @JohnJMartins! I've requested some changes based on some typos/terminology.

It also seems as if some of the panel rows do not span full-width, do you think you'd be able to adjust them to be full-width (no big deal if this isn't feasible).

**Comment by JohnJMartins** [comment]
> Thank you for the contribution @JohnJMartins! I've requested some changes based on some typos/terminology.
> 
> It also seems as if some of the panel rows do not span full-width, do you think you'd be able to adjust them to be full-width (no big deal if this isn't feasible).

Hi Andreas, it is not very feasible to have the panel widths extended for this dashboard so I have foregone that change, but I have fixed the issues mentioned in other comments. Please let me know if there is anything else you need me to do!

**Review by aangelisc — APPROVED** [review]
Thank you for updating @JohnJMartins. LGTM!

**Pull request merged**